File: c01d-br0th3r/0318/7568.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Person {
    int height;
    int weight;
    int rank;
} Person;

int main() {
    Person *p, tmp;
    int i, j, N;
    
    scanf("%d", &N);
    
    p = (Person *)malloc(sizeof(Person) * N);
    
    for (i = 0; i < N; i++) {
        scanf("%d %d", &p[i].weight, &p[i].height);
        p[i].rank = 1;
    }
    
    for (i = 0; i < N; i++) {
        for (j = 0; j < N; j++) {
            if (p[i].height < p[j].height && p[i].weight < p[j].weight) {
                p[i].rank++;
            }
        }
    }
    
    for (i = 0; i < N; i++) {
        printf("%d ", p[i].rank);
    }
    
    free(p);
    
    return 0;
}

File: hyeon-uk/2021/1101~1107/7568.java
import java.util.*;

public class Main {
    static Vector<Person> v=new Vector<>();
    static int n;

    static class Person implements Comparable<Person>{
        int weight;
        int height;

        public Person(int weight, int height) {
            this.weight = weight;
            this.height = height;
        }


        @Override
        public int compareTo(Person o) {
            if(this.weight<o.weight&&this.height<o.height){
                return 1;
            }
            else{
                return 0;
            }
        }
    }

    public static void main(String[] args){
        Scanner scanner=new Scanner(System.in);
        n=scanner.nextInt();

        for(int i=0;i<n;i++){
            int w=scanner.nextInt();
            int h=scanner.nextInt();

            v.add(new Person(w,h));
        }
        for(int i=0;i<n;i++){
            int rank=1;
            for(int j=0;j<n;j++){
                if(i==j) continue;
                if(v.get(i).compareTo(v.get(j))==1) rank++;
            }
            System.out.print(rank+" ");
        }
    }

}
